### PR TITLE
Enhance query_balance.py

### PR DESCRIPTION
**Pull request by hanyunxu**
For #591, add query balance by
- recipient 
- minor block height

### CODE REVIEW

**Comment by ninjaahhh** [comment]
1. need to support querying a recipient's balances in all shards
1. need to support querying a shard's balance at a specific height

File: quarkchain/tools/query_balance.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--full_shard_id", type=int, help="full shard id to operate")
+    parser.add_argument(
+        "--all_shards",
+        action="store_true",
+        default=False,
+        help="query balances in all shards",
+    )
+    parser.add_argument("--recipient", type=int, help="query a specific recipient")
```
Comment (qizhou) [inline quarkchain/tools/query_balance.py]: default value?

File: quarkchain/tools/query_balance.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    parser.add_argument(
```
Comment (qizhou) [inline quarkchain/tools/query_balance.py]: default value?

**Comment by hanyunxu** [comment]
> It is also possible to query a balance of a recipient at a specific height?

It is included in `print_recipient_balance()` [L105](https://github.com/QuarkChain/pyquarkchain/blob/56c7c5e70e423fe6f06cd86a4cbac51861f935e5/quarkchain/tools/query_balance.py#L105)

File: quarkchain/tools/query_balance.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    state = shard.state
+    state.init_from_root_block(rb)
+
+    if args.minor_block_height:
+        balance = state.get_balances(recipient, args.minor_block_height)
+    else:
+        balance = state.get_balances(recipient)
+
+    if balance:
+        print("Recipient: %d，balance: %d", (recipient, balance))
```
Comment (qizhou) [inline quarkchain/tools/query_balance.py]: missing format %

**Pull request merged**